OSDOCS-12533: ABI on OCI GA (#84551)
openshift/openshift-docs: modules/creating-config-files-cluster-install-oci.adoc

# ...
----
<1> The base domain of your cloud provider.
<2> The IP address from the virtual cloud network (VCN) that the CIDR allocates to resources and components that operate on your network.
<3> Depending on your infrastructure, you can select either `arm64` or `amd64`.
<4> Set `OCI` as the external platform, so that {product-title} can integrate with {oci}.
<5> Specify your SSH public key.
<6> The pull secret that you need for authenticate purposes when downloading container images for {product-title} components and services, such as Quay.io. See link:https://console.redhat.com/openshift/install/pull-secret[Install {product-title} 4] from the {hybrid-console}.

. Create a directory on your local system named `openshift`. This must be a subdirectory of the installation directory.
+
[IMPORTANT]
====

> Do not move the `install-config.yaml` and `agent-config.yaml` configuration files to the `openshift` directory.

Reviewer: out-of-scope nit that I noticed
```suggestion
Do not move the `install-config.yaml` or `agent-config.yaml` configuration files to the `openshift` directory.
```

====

. If you used a stack to provision OCI infrastructure resources: Copy and paste the `dynamic_custom_manifest` output of the OCI stack into a file titled `manifest.yaml` and save the file in the `openshift` directory.

. If you did not use a stack to provision OCI infrastructure resources: Download and prepare custom manifests to create an Agent ISO image:

.. Go to link:https://docs.oracle.com/iaas/Content/openshift-on-oci/install-prereq.htm#install-configuration-files[Configuration Files] (Oracle documentation) and follow the link to the custom manifests directory on GitHub.

.. Copy the contents of the `condensed-manifest.yml` file and save it locally to a file in the `openshift` directory.

> .. In the `condensed-manifest.yml` file, update the sections marked with `TODO` to specify the compartment {ocid}, VCN {ocid}, subnet {ocid} from the load balancer, and the security lists {ocid}.

Reviewer: I don't see this defined in the assembly
```suggestion
.. In the `condensed-manifest.yml` file, update the sections marked with `TODO` to specify the compartment {ocid-first}, VCN {ocid}, subnet {ocid} from the load balancer, and the security lists {ocid}.
```

. Configure the `agent-config.yaml` configuration file to meet your organization's requirements.
+

> .`agent-config.yaml` file that sets values for an IPv4 formatted network.

Reviewer: Doesn't need a `.`, probably should call it a "sample", and I _think_ you could shorten it with no impact to understanding 
```suggestion
.Sample `agent-config.yaml` file for an IPv4 network
```

Author: Good call, not my original writing (i only switched the order of steps for accuracy) so I'm happy to trim out any excess

[source,yaml]
----
apiVersion: v1beta1
metadata:

>   name: <cluster_name> // <1>

Reviewer: AFAIK you don't need this in YAML blocks (you could use `#` but it shouldn't matter)
```suggestion
  name: <cluster_name> <1>
```

  namespace: <cluster_namespace> <2>
rendezvousIP: <ip_address_from_CIDR> <3>
bootArtifactsBaseURL: <server_URL> <4>
# ...
----
<1> The cluster name that you specified in your DNS record.
<2> The namespace of your cluster on {product-title}.

> <3> If you use IPv4 as the network IP address format, ensure that you set the `rendezvousIP` parameter to an IPv4 address that the VCN's Classless Inter-Domain Routing (CIDR) method allocates on your network. Also ensure that at least one instance from the pool of instances that you booted with the ISO matches the IP address value you set for `rendezvousIP`.

Reviewer: ```suggestion
<3> If you use IPv4 as the network IP address format, ensure that you set the `rendezvousIP` parameter to an IPv4 address that the VCN's Classless Inter-Domain Routing (CIDR) method allocates on your network. Also ensure that at least one instance from the pool of instances that you booted with the ISO matches the IP address value you set for the `rendezvousIP` parameter.
```

> Do not move the `install-config.yaml` and `agent-config.yaml` configuration files to the `openshift` directory.

Author: In the step on [line 97](https://github.com/openshift/openshift-docs/pull/84551/files#diff-69efedd831fae26c26fdbda29b3ef4d1eef22ac76460b6f6cec52f4bd62ab949R97): If it was valid to add a step about creating an installation directory earlier, should this `openshift` directory be a subdirectory of that?

Reviewer 2: Yes, it would be a subdirectory of the installation directory. Also, remove the extra space in step 2 between `the` and `virtual cloud network (VCN)` and in step 3, change it to `amd64` or `arm64`.

====

. Download and prepare custom manifests for Oracle:

.. Go to link:https://docs.oracle.com/iaas/Content/openshift-on-oci/install-prereq.htm#install-configuration-files[Configuration Files] (Oracle documentation).

.. Download the {oci-ccm-full} and {oci-csi-full} manifests as an archive file and save the archive file in your `openshift` directory.
+
You need the {oci-ccm} manifests for deploying the {oci-ccm} during cluster installation so that {product-title} can connect to the external {oci} platform. You need the {oci-csi} custom manifests for deploying the {oci-csi} driver during cluster installation so that {product-title} can claim required objects from {oci}.

.. Access the {oci-ccm} and {oci-csi} manifests that you downloaded.

.. Change the `oci-cloud-controller-manager` secret that is defined in the `oci-ccm.yml` configuration file to match your organization's region, compartment {ocid}, VCN {ocid}, and the subnet {ocid} from the load balancer.

> . Generate a minimal ISO image, which excludes the rootfs image, by entering the following command. You can use this image later in the process to boot all your cluster's nodes.

Author: Similar to my other question - if adding the installation directory step was valid, should this command have an argument to specify that dir? Or, alternatively, should I mention in the wording that you need to run this in your installation directory?

Reviewer 2: Either option would be fine.

Author: Sounds good. I think I might also delete the second sentence since it's only telling users about an action they're going to take later anyways, and otherwise it's interrupting the flow of the step.

Do not move the `install-config.yaml` and `agent-config.yaml` configuration files to the `openshift` directory.
====

. If you used a stack to provision OCI infrastructure resources: Copy and paste the `dynamic_custom_manifest` output of the OCI stack into a file titled `manifest.yaml` and save the file in the `openshift` directory.

> . If you did not use a stack to provision OCI infrastructure resources: Download and prepare custom manifests for Oracle.

Reviewer 2: ```suggestion
. If you did not use a stack to provision OCI infrastructure resources: Download and prepare custom manifests to create an Agent ISO image.
```
By the way, it should not be a step, but rather a note in the above step.

Author: This might be tough to do because this step introduces and nests the substeps below that talk about downloading and manually configuring the custom manifests, and we can't really have steps in a note. If you want me to make that relationship more clear, I could maybe end the sentence with a colon like this:
```suggestion
. If you did not use a stack to provision OCI infrastructure resources: Download and prepare custom manifests for to create an Agent ISO image:
```

.. Go to link:https://docs.oracle.com/iaas/Content/openshift-on-oci/install-prereq.htm#install-configuration-files[Configuration Files] (Oracle documentation).

> .. Download the {oci-ccm-full} and {oci-csi-full} manifests as an archive file and save the archive file in your `openshift` directory.

Reviewer 2: Currently, there are 8 files under the Oracle repo that need to be placed in the OpenShift directory. The user needs to edit the `01-oci-driver-configs.yml` file to update the compartment OCID, VCN OCID, subnet OCID from the load balancer and security lists OCID.

I would suggest getting in touch with Oracle folks for more information about it.

Author: I can ask an Oracle engineer for feedback on making this overall part more accurate

+
You need the {oci-ccm} manifests for deploying the {oci-ccm} during cluster installation so that {product-title} can connect to the external {oci} platform. You need the {oci-csi} custom manifests for deploying the {oci-csi} driver during cluster installation so that {product-title} can claim required objects from {oci}.

.. Access the {oci-ccm} and {oci-csi} manifests that you downloaded.

.. Change the `oci-cloud-controller-manager` secret that is defined in the `oci-ccm.yml` configuration file to match your organization's region, compartment {ocid}, VCN {ocid}, and the subnet {ocid} from the load balancer.

. Configure the `agent-config.yaml` configuration file to meet your organization's requirements.
+
.`agent-config.yaml` file that sets values for an IPv4 formatted network.
[source,yaml]
----

> apiVersion: v1alpha1

Reviewer 2: This can be updated to v1beta1.

+
[IMPORTANT]
====
The Agent-based Installer uses {op-system-first}. The rootfs image, which is mentioned in a later step, is required for booting, recovering, and repairing your operating system.
====

. Disconnected environments only: Upload the rootfs image to a web server.

..  Go to the `./<installation_directory>/boot-artifacts` directory that was generated when you created the minimal ISO image.

.. Use your preferred web server, such as any Hypertext Transfer Protocol daemon (`httpd`), to upload the rootfs image to the location specified in the `bootArtifactsBaseURL` parameter of the `agent-config.yaml` file.
+

> For example, if the `bootArtifactsBaseURL` parameter states `\http://192.168.122.20`, you would upload the generated rootfs image to this location, so that the Agent-based installer can access the image from `\http://192.168.122.20/agent.x86_64-rootfs.img`. After the Agent-based installer boots the minimal ISO for the external platform, the Agent-based Installer downloads the rootfs image from the `\http://192.168.122.20/agent.x86_64-rootfs.img` location into the system memory.

Reviewer: out-of-scope nit
```suggestion
For example, if the `bootArtifactsBaseURL` parameter states `\http://192.168.122.20`, you would upload the generated rootfs image to this location so that the Agent-based installer can access the image from `\http://192.168.122.20/agent.x86_64-rootfs.img`. After the Agent-based installer boots the minimal ISO for the external platform, the Agent-based Installer downloads the rootfs image from the `\http://192.168.122.20/agent.x86_64-rootfs.img` location into the system memory.
```

+

> [NOTE]

Reviewer: _If possible_, try to avoid stacking admonitions.

Author: In a later effort I will consider whether this Note text even needs to be in a note rather than in a paragraph
